src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -760,12 +760,11 @@ static int enumerate_partitions(dev_t devnum) {
                          * on. And thus we also don't set DISSECT_IMAGE_PIN_PARTITION_DEVICES here, because
                          * we don't actually mount anything immediately. */
                         &m);
-        if (r == -ENOPKG) {
-                log_debug_errno(r, "No suitable partition table found on block device %s, ignoring.", devname);
-                return 0;
+        if (r < 0) {
+                bool ok = IN_SET(r, -ENOPKG, -ENOMSG);
```

Comment:
After https://github.com/systemd/systemd/commit/ee0e6e476e61d4baa2a18e241d212753e75003bf, `-ENOMSG` should not be returned. Please drop it.